lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -238,5 +252,289 @@ bool circt::firrtl::fromJSON(json::Value &value,
+  //   goto unimplemented;
+
+  // An AugmentedBundleType looks like:
+  //   "defName": String
+  //   "elements": Seq[AugmentedField]
+  if (clazz.getValue() == (Twine(package) + ".AugmentedBundleType").str()) {
+    defName = augmentedType.get("defName").cast<StringAttr>();
+
+    // Each element is an AugmentedField with members:
+    //   "name": String
+    //   "description": Option[String]
+    //   "tpe": AugmenetedType
+    SmallVector<Attribute> elements;
+    for (auto elt : augmentedType.get("elements").cast<ArrayAttr>()) {
+      auto field = elt.cast<DictionaryAttr>();
+      auto name = field.get("name").cast<StringAttr>();
+
+      auto tpe = field.get("tpe").cast<DictionaryAttr>();
+      parseAugmentedType(context, tpe, newAnnotations, companion, name,
+                         defName);
+
+      // Collect information necessary to build a module with this view later.
+      // This includes the optional description and name.
+      NamedAttrList attrs;
+      if (auto maybeDescription = field.get("description"))
+        attrs.append("description", maybeDescription.cast<StringAttr>());
+      attrs.append("name", name);
+      elements.push_back(DictionaryAttr::getWithSorted(context, attrs));
+    }
+    // Add an annotation that stores information necessary to construct the
+    // module for the view.  This needs the name of the module (defName) and the
+    // names of the components inside it.
+    NamedAttrList attrs;
+    attrs.append("class", clazz);
+    attrs.append("defName", defName);
+    attrs.append("elements", ArrayAttr::get(context, elements));
+    newAnnotations["~"].push_back(
+        DictionaryAttr::getWithSorted(context, attrs));
+    return;
+  }
+
+  // An AugmentedGroundType looks like:
+  //   "ref": ReferenceTarget
+  //   "tpe": GroundType
+  // The ReferenceTarget is not serialized to a string.  The GroundType will
+  // either be an actuall FIRRTL ground type or a GrandCentral uninferred type.
+  // This can be ignored for us.
+  if (clazz.getValue() == (Twine(package) + ".AugmentedGroundType").str()) {
+    auto target =
+        refTargetToString(augmentedType.get("ref").cast<DictionaryAttr>());
+    NamedAttrList attr;
+    attr.append("class", clazz);
+    attr.append("defName", defName);
+    attr.append("name", name);
+    if (target.second)
+      attr.append("target", target.second);
+    newAnnotations[target.first].push_back(DictionaryAttr::get(context, attr));
+    return;
+  }
+
+  // An AugmentedVectorType looks like:
+  //   "elements": Seq[AugmentedType]
+  if (clazz.getValue() == (Twine(package) + ".AugmentedVectorType").str()) {
+    for (auto elt : augmentedType.get("elements").cast<ArrayAttr>())
+      parseAugmentedType(context, elt.cast<DictionaryAttr>(), newAnnotations,
+                         companion, name, defName);
+    return;
+  }
+
+  llvm::errs() << "Handling of '" << clazz << "' is not implemented yet!\n";
+  llvm_unreachable("Code not implemented");
```

Comment:
This is reachable with a malformed attr, please change it into a return and handle the error case cleanly without crashing.

Reply:
Fixed in a later commit. All the errors are now handled. Any of the following are now caught and fail cleanly:
1. Annotation field is not found
2. Field type does not match
3. A class is not able to be handled

E.g.:

```
gct/MemTapBadTap.fir:1:15: error: Annotation 'sifive.enterprise.grandcentral.MemTapAnnotation' with path '.taps[0]' contained an unexpected type (expected a string).
circuit Foo : %[
              ^
gct/MemTapBadTap.fir:1:15: note: The full Annotation is reprodcued here: {class = "sifive.enterprise.grandcentral.MemTapAnnotation", source = "~Foo|Foo>mem", taps = [{foo = "bar"}]}
```

Reply:
😂 Well, I thought this was fixed... Apparently not.

Reply:
Actually fixed in a later commit. This will now produce:

```
gct/InvalidAugmentedTypeTarget.fir:1:24: error: Unknown AugmentedType 'sifive.some.unsupported.thing'. Either this is unsupported or maybe you mispelled it?
circuit GCTInterface : %[
                       ^
```